File: solutions/c/collatz-conjecture/1/collatz_conjecture.c
#include "collatz_conjecture.h"

int steps(int start)
{
    if (start <= 0)
        return -1;
    int i = 0;
    int num = start;
    while (num > 1)
    {
        if (num % 2)
            num = 3 * num + 1;
        else
            num /= 2;
        i++;
    }
    return i;
}

File: solutions/fsharp/collatz-conjecture/1/CollatzConjecture.fs
module CollatzConjecture

let steps (number: int): int option = 
    let mutable counter = 0 |> Some
    if number > 0 then
        let mutable n = number
        while n <> 1 do
            n <- if n % 2 = 0 then n / 2 else 3 * n + 1
            counter <- Option.map ((+) 1) counter
        counter
    else
        None